model/SyncLog/SyncLogFilter.php

```diff
@@ -0,0 +1,286 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoSync\model\SyncLog;
+
+use oat\taoSync\model\SyncLog\Storage\SyncLogStorageInterface;
+
+/**
+ * Class SyncLogFilter
+ * @package oat\taoSync\model\SyncLog
+ */
+class SyncLogFilter
+{
+    const OP_EQ  = '=';
+    const OP_NEQ = '!=';
+    const OP_LT  = '<';
+    const OP_LTE = '<=';
+    const OP_GT  = '>';
+    const OP_GTE = '>=';
+    const OP_LIKE = 'LIKE';
+    const OP_NOT_LIKE = 'NOT LIKE';
+    const OP_IN = 'IN';
+    const OP_NOT_IN = 'NOT IN';
+
+    private $filters = [];
+    private $limit;
+    private $offset;
+    private $sortBy;
+    private $sortOrder;
+    private $columns = [
+        SyncLogStorageInterface::COLUMN_ID,
+        SyncLogStorageInterface::COLUMN_BOX_ID,
+        SyncLogStorageInterface::COLUMN_SYNC_ID,
+        SyncLogStorageInterface::COLUMN_ORGANIZATION_ID,
+        SyncLogStorageInterface::COLUMN_DATA,
+        SyncLogStorageInterface::COLUMN_STATUS,
+        SyncLogStorageInterface::COLUMN_REPORT,
+        SyncLogStorageInterface::COLUMN_STARTED_AT,
+        SyncLogStorageInterface::COLUMN_FINISHED_AT,
+    ];
+
+    /**
+     * @return array
+     */
+    public function getColumns()
+    {
+        return $this->columns;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getSortBy()
+    {
+        return $this->sortBy;
+    }
+
+    /**
+     * @param mixed $sortBy
+     * @return SyncLogFilter
+     */
+    public function setSortBy($sortBy)
+    {
+        $this->sortBy = $sortBy;
+        return $this;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getSortOrder()
+    {
+        return $this->sortOrder;
+    }
+
+    /**
+     * @param mixed $sortOrder
+     * @return SyncLogFilter
+     */
+    public function setSortOrder($sortOrder)
+    {
+        $this->sortOrder = $sortOrder;
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLimit()
+    {
+        return $this->limit;
+    }
+
+    /**
+     * @param int $limit
+     * @return SyncLogFilter
+     */
+    public function setLimit($limit)
+    {
+        $this->limit = max(0, $limit);
+        return $this;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOffset()
+    {
+        return $this->offset;
+    }
+
+    /**
+     * @param int $offset
+     * @return SyncLogFilter
+     */
+    public function setOffset($offset)
+    {
+        $this->offset = max(0, $offset);
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function getFilters()
+    {
+        return $this->filters;
+    }
+
+    /**
+     * @param string $field
+     * @param string $operator
+     * @param mixed $value
+     * @return SyncLogFilter
+     */
+    public function addFilter($field, $operator, $value)
```

Comment:
All filters will be combined by `AND` operator?
So it will not be possible to find all entities where box id = 10 `OR` organization id = 10?

Reply:
Yes, you are right. It's not necessary right now, but would be useful to fix in future.